Improve dm= command: unified colors, unicode support, and alignment fix

Ayushd785 opened the pull request:
## Summary
This PR enhances the `dm=` command output to provide a more consistent and visually appealing experience for memory map visualization.

## Changes Made
- **Unicode Support**: The command now uses Unicode block characters (█) and horizontal lines (─) when `scr.utf8` is enabled, falling back to ASCII characters (# and -) when disabled
- **Unified Color Scheme**: All elements (addresses, bars, vertical separators, and permission text) now use the same color for each memory map based on its permissions
- **Alignment Fix**: Fixed the size column formatting from `%4.8s` to `%5s` to ensure consistent alignment across different size values
- **Enhanced Visual Design**: Permission text and vertical separators are now colored to create a cohesive visual connection

## Screenshots-

before- 
<img width="1920" height="374" alt="image" src="https://github.com/user-attachments/assets/730938bb-df6d-4824-8b36-d35fef687406" />

After- 
<img width="1920" height="434" alt="image" src="https://github.com/user-attachments/assets/7b69ad85-ca54-47c6-a461-33115afe1de1" />


## Color Mapping
- **RED** (`ai.exec`) for executable memory regions (`r-x`, `--x`)
- **BLUE** (`ai.write`) for writable memory regions (`rw-`)
- **GREEN** (`ai.read`) for read-only memory regions (`r--`)
- **BRIGHT RED** (`widget_sel`) for dangerous memory regions with both write and execute permissions (`rwx`)


Fixes #5251

Ayushd785 committed:
Improved dm= command: unified colors, unicode support, and alignment fix.

Ayushd785 commented:
## AI declaration 

I have used GitHub Copilot for any syntax issues and Gemini for drafting and discussions. There is no line of code written by an AI in this pull request. Most of the code is just reused from other implementations .
like - Unicode Character pattern 

`const char *block = opts->unicode ? UTF_BLOCK : "#";`

for color palette usage-

`if (type & RZ_ANALYSIS_ADDR_TYPE_EXEC) {
    return core->cons->context->pal.ai_exec; // Color_RED;
}
if (type & RZ_ANALYSIS_ADDR_TYPE_WRITE) {
    return core->cons->context->pal.ai_write; // Color_BLUE;
}
if (type & RZ_ANALYSIS_ADDR_TYPE_READ) {
    return core->cons->context->pal.ai_read; // Color_GREEN;
}`

Ayushd785 committed:
Merge branch 'dev' into improve-dm-command-colors-unicode

Ayushd785 committed:
tests: add dm= unicode+color test for coverage

Ayushd785 commented:
Tested with larger memory maps using Rizin’s Cutter and Firefox to trigger them.
Cutter-
<img width="1919" height="260" alt="Screenshot From 2025-10-23 12-40-56" src="https://github.com/user-attachments/assets/cbe07bff-76c6-48a4-bdaf-135a4f2863ba" />
Firefox-
<img width="1920" height="258" alt="Screenshot From 2025-10-23 12-44-49" src="https://github.com/user-attachments/assets/65c88724-1166-4c65-8bcc-d88e1ed1a535" />

Ayushd785 committed:
tests: remove dm= unicode+color tests as we can't test the whole output of the dm because it differs between environments

Ayushd785 commented:
@notxvilka As per the suggestion, the tests for the `dm=` command have been completely removed because the `~ `command does not work as expected for output filtering with dm=, and the visual output varies across environments, making it unsuitable for automated testing.

notxvilka commented on librz/core/cdebug.c:
```diff
@@ -668,29 +678,45 @@ static void print_debug_maps_ascii_art(RzDebug *dbg, RzList /*<RzDebugMap *>*/ *
+				colors && bar_color[0] ? bar_color : "",
+				colors && bar_color[0] ? Color_RESET : ""); // * indicates map is within our current sought offset
 			int col;
-			for (col = 0; col < width; col++) { // Iterate over the available width/columns for bar graph
```
Please do not remove old comments. Apart from that LGTM

Ayushd785 replied:
i have restored the old comments some of them didn't made any sense for the new code so i added new ones for them

notxvilka commented on test/db/archos/linux-x64/dbg_maps:
```diff
@@ -9,6 +9,7 @@ EXPECT=<<EOF
 EOF
 RUN
 
+
```
Unnecessary change

Ayushd785 replied:
i have removed the extra line of change and kept only one line of space between those two to keep the test structure consistent as it was

Ayushd785 commented:
I sincerely apologize for the extra force-pushes. I was stuck with some Git synchronization issues where changes weren't being properly reflected in commits, even though they appeared correctly in my working directory. I had removed the extra line from dbg_maps, but it wasn't showing up in the committed changes. There was something I was missing, but I've fixed it now.

notxvilka commented on librz/core/cdebug.c:
```diff
@@ -644,20 +647,27 @@ static void print_debug_maps_ascii_art(RzDebug *dbg, RzList /*<RzDebugMap *>*/ *
 		const char *fmtstr;
 		char humansz[8]; // Holds the human formatted size string [124K]
-		int skip = 0; // Number of maps to skip when re-calculating the minmax
+		int skip = 0;  // Number of maps to skip when re-calculating the minmax
```
One unnecessary space character here

Ayushd785 replied:
Yes sorry for that, Its fixed now

notxvilka reviewed (changes requested):
Just one little thing left
